modules/core/src/transitions/viewport-fly-to-interpolator.js

```diff
@@ -18,42 +14,19 @@ const LINEARLY_INTERPOLATED_PROPS = ['bearing', 'pitch'];
  */
 export default class ViewportFlyToInterpolator extends TransitionInterpolator {
   constructor() {
-    super();
-    this.propNames = VIEWPORT_TRANSITION_PROPS;
-  }
-
-  initializeProps(startProps, endProps) {
-    const startViewportProps = {};
-    const endViewportProps = {};
-
-    // Check minimum required props
-    for (const key of REQUIRED_PROPS) {
-      const startValue = startProps[key];
-      const endValue = endProps[key];
-      assert(isValid(startValue) && isValid(endValue), `${key} must be supplied for transition`);
-      startViewportProps[key] = startValue;
-      endViewportProps[key] = getEndValueByShortestPath(key, startValue, endValue);
-    }
-
-    for (const key of LINEARLY_INTERPOLATED_PROPS) {
-      const startValue = startProps[key] || 0;
-      const endValue = endProps[key] || 0;
-      startViewportProps[key] = startValue;
-      endViewportProps[key] = getEndValueByShortestPath(key, startValue, endValue);
-    }
-
-    return {
-      start: startViewportProps,
-      end: endViewportProps
-    };
+    super({
+      compare: ['longitude', 'latitude', 'zoom', 'bearing', 'pitch'],
+      extract: ['width', 'height', 'longitude', 'latitude', 'zoom', 'bearing', 'pitch'],
+      required: ['width', 'height', 'latitude', 'longitude', 'zoom']
+    });
   }
 
   interpolateProps(startProps, endProps, t) {
     const viewport = flyToViewport(startProps, endProps, t);
 
     // Linearly interpolate 'bearing' and 'pitch' if exist.
     for (const key of LINEARLY_INTERPOLATED_PROPS) {
-      viewport[key] = lerp(startProps[key], endProps[key], t);
+      viewport[key] = lerp(startProps[key] || 0, endProps[key] || 0, t);
```

Comment:
Nit: Are those good fallbacks or just to avoid crash/garbage values? Better to not interpolate if these are not defined in both places? Maybe a comment or TODO? Worth a test case to trigger?

Reply:
If pitch/bearing are not defined, they are interpreted as zeros in viewport calculation (fallback defined in `WebMercatorViewport`). Therefore we still need to transition them.

Reply:
OK thanks.

Might this be worth a comment in the code rather than the PR?
in a perfect world, I suppose the same system/code should be used to calculate fallbacks in both casesa?